OutOfSchool/OutOfSchool.IdentityServer/Data/Migrations/OutOfSchoolMigrations/20210828133338_ActualAddressCanBeNull.Designer.cs

```diff
@@ -0,0 +1,1286 @@
+                {
+                    b.HasOne("OutOfSchool.Services.Models.Parent", "Parent")
+                        .WithMany("Children")
+                        .HasForeignKey("ParentId")
+                        .OnDelete(DeleteBehavior.Cascade)
+                        .IsRequired();
+
+                    b.HasOne("OutOfSchool.Services.Models.SocialGroup", "SocialGroup")
+                        .WithMany("Children")
+                        .HasForeignKey("SocialGroupId");
+
+                    b.Navigation("Parent");
+
+                    b.Navigation("SocialGroup");
+                });
+
+            modelBuilder.Entity("OutOfSchool.Services.Models.Class", b =>
+                {
+                    b.HasOne("OutOfSchool.Services.Models.Department", "Department")
+                        .WithMany("Classes")
+                        .HasForeignKey("DepartmentId")
+                        .OnDelete(DeleteBehavior.Cascade)
+                        .IsRequired();
+
+                    b.Navigation("Department");
+                });
+
+            modelBuilder.Entity("OutOfSchool.Services.Models.Department", b =>
+                {
+                    b.HasOne("OutOfSchool.Services.Models.Direction", "Direction")
+                        .WithMany("Departments")
+                        .HasForeignKey("DirectionId")
+                        .OnDelete(DeleteBehavior.Cascade)
+                        .IsRequired();
+
+                    b.Navigation("Direction");
+                });
+
+            modelBuilder.Entity("OutOfSchool.Services.Models.Favorite", b =>
+                {
+                    b.HasOne("OutOfSchool.Services.Models.User", "User")
+                        .WithMany()
+                        .HasForeignKey("UserId")
+                        .OnDelete(DeleteBehavior.Cascade)
+                        .IsRequired();
+
+                    b.HasOne("OutOfSchool.Services.Models.Workshop", "Workshop")
+                        .WithMany()
+                        .HasForeignKey("WorkshopId")
+                        .OnDelete(DeleteBehavior.Cascade)
+                        .IsRequired();
+
+                    b.Navigation("User");
+
+                    b.Navigation("Workshop");
+                });
+
+            modelBuilder.Entity("OutOfSchool.Services.Models.Parent", b =>
+                {
+                    b.HasOne("OutOfSchool.Services.Models.User", "User")
+                        .WithMany()
+                        .HasForeignKey("UserId")
+                        .OnDelete(DeleteBehavior.Cascade)
+                        .IsRequired();
+
+                    b.Navigation("User");
+                });
+
+            modelBuilder.Entity("OutOfSchool.Services.Models.Provider", b =>
+                {
+                    b.HasOne("OutOfSchool.Services.Models.Address", "ActualAddress")
+                        .WithMany()
+                        .HasForeignKey("ActualAddressId");
```

Comment:
I think we need to add ` .OnDelete(DeleteBehavior.Cascade)`
Now when we delete Provider it does not delete the Actual address from DB.

Reply:
I will update provider model creation in dbContext later so EF will be able to prepare its migration better